extensions/data-workspace/src/common/telemetry.ts

```diff
@@ -42,3 +35,22 @@ export function calculateRelativity(projectPath: string, workspacePath?: string)
 
 	return 'other'; // sibling, cousin, descendant, etc.
 }
+
+
+export enum TelemetryViews {
+	WorkspaceTreePane = 'WorkspaceTreePane',
+	OpenExistingDialog = 'OpenExistingDialog',
+	NewProjectDialog = 'NewProjectDialog',
+	ProviderRegistration = 'ProviderRegistration'
+}
+
+export enum TelemetryActions {
+	ProviderRegistered = 'ProviderRegistered',
+	ProjectAddedToWorkspace = 'ProjectAddedToWorkspace',
+	ProjectRemovedFromWorkspace = 'ProjectRemovedFromWorkspace',
+	OpeningProject = 'OpeningProject',
+	NewProjectDialogLaunched = 'NewProjectDialogLaunched',
+	OpeningWorkspace = 'OpeningWorkspace',
+	OpenWorkspaceProjectDialogLaunched = 'OpenWorkspaceProjectDialogLaunched',
```

Comment:
would be shorter to call this OpenExistingDialogLaunched. Also would be good to change it since we'll add opening from github in this dialog soon.